bot.go

```diff
@@ -116,6 +132,16 @@ func NewBot(confReader io.Reader) (*Bot, error) {
 		b.log.Logger.Level = logrus.DebugLevel
 	}
 
+	// Set up InfluxDB logging
+	err = b.Config("influxdb", &b.influxDbConfig)
+	if err == nil {
+		b.lastPointSubmit = time.Now()
+		b.points = list.New()
```

Comment:
b.points would probably work best as a channel. You can have a buffered channel (`someChan := make(chan someType, maxSize)`) and can write to that concurrently or drop a message like this:

```
switch {
case someChan <- someValue:
default:
log.Println("Dropping value")
}
```

Then your writer would just wait for `val := <-someChan` (or the bot to exit, which I'm not 100% sure how that works - it's been a while since I've looked)